### PR TITLE
[WOR-1353] Update the workspace's auth domain on PAO link

### PR DESCRIPTION
**Pull request by aherbst-broad**
[Related ticket](https://broadworkbench.atlassian.net/browse/WOR-1353)

We are not setting the auth domain on a workspace when a data repo snapshot is imported with a TPS group-constraint policy. This PR updates the linkPoilcies method to add the group from the constraint to the workspace's auth domain, similar to what we're doing on policy updates.

### CODE REVIEW

**Comment by aherbst-broad** [comment]
Azure tests are failing due to the IT MRG subnet being full, which is unrelated. Looking into that but circulating now to get eyes on this change.

**Review by marctalbott — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java [hunk 1]
```diff
@@ -859,6 +870,8 @@ public TpsPaoUpdateResult linkPolicies(
         workspaceId,
         sourcePaoId.getObjectId(),
         userRequest.getEmail());
+    TpsPaoGetResult paoBeforeUpdate =
+        Rethrow.onInterrupted(() -> tpsApiDispatch.getPao(workspaceId), "getPao");
```
Comment (cahrens) [inline service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java:874]: Should you use `getOrCreatePao` just be safe?

File: service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java [hunk 2]
```diff
@@ -968,23 +984,31 @@ public TpsPaoUpdateResult updatePolicy(
...
-                    addedGroups.stream().toList()),
-            "updateAuthDomains");
-      }
+      patchWorkspaceAuthDomain(workspaceUuid, userRequest, paoBeforeUpdate, dryRun);
 
       return result;
     }
   }
+
+  private void patchWorkspaceAuthDomain(
```
Comment (cahrens) [inline service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java:993]: Do auth domain groups ever get removed (like in the case of having to undo a step)?

**Review by cahrens — APPROVED** [review]

**Comment by cahrens** [comment]
@aherbst-broad is this good to merge?

**Pull request merged**